test/Datadog.Trace.IntegrationTests/Datadog.Trace.IntegrationTests.csproj

```diff
@@ -11,5 +11,23 @@
 
   <ItemGroup Condition=" !$(TargetFramework.StartsWith('net4')) ">
     <PackageReference Include="System.Net.Http" Version="4.3.4" />
+    <ProjectReference Include="..\test-applications\integrations\Samples.AspNetCoreRazorPages\Samples.AspNetCoreRazorPages.csproj" />
+  </ItemGroup>
+
+  <ItemGroup Condition=" $(TargetFramework.StartsWith('netcoreapp2')) ">
+    <PackageReference Include="Microsoft.AspNetCore" Version="2.2.0" />
+    <PackageReference Include="Microsoft.AspNetCore.Mvc" Version="2.2.0" />
+    <PackageReference Include="Microsoft.AspNetCore.Mvc.Testing" Version="2.2.0" />
+    <PackageReference Include="Microsoft.AspNetCore.TestHost" Version="2.2.0" />
+  </ItemGroup>
+
+  <ItemGroup Condition=" $(TargetFramework.StartsWith('netcoreapp3')) ">
+    <PackageReference Include="Microsoft.AspNetCore.Mvc.Testing" Version="3.0.0" />
+    <PackageReference Include="Microsoft.AspNetCore.TestHost" Version="3.0.0" />
+  </ItemGroup>
+
+  <ItemGroup Condition=" $(TargetFramework.StartsWith('net5')) ">
+    <PackageReference Include="Microsoft.AspNetCore.Mvc.Testing" Version="5.0.0" />
+    <PackageReference Include="Microsoft.AspNetCore.TestHost" Version="5.0.0" />
```

Comment:
I'm not a fan of adding all these dependencies to `Datadog.Trace.IntegrationTests`, including `Microsoft.NET.Sdk.Web` and the `Samples.AspNetCoreRazorPages` project. Makes me want to move all these tests into a separate project. Maybe something we can come back and clean up later, no need to change now.

Reply:
> I'm not a fan of adding all these dependencies

Any particular reason? This project has _very_ few tests currently (8 tests. 6 of which are skipped). The project reference isn't ideal, but it's basically the only way to test Razor Pages properly, short of spinning up things in IIS etc (which obviously doesn't work on Linux unless we switch to raw kestrel instead). Plus these tests are an order of magnitude faster than using IIS...

That said, we've already got 106 projects... what's 1 more 😬

Reply:
> Any particular reason?

I'm used to having one test assembly for every assembly (`{assembly}.Tests`) to keep inter-dependencies to a minimum:
- `Datadog.Trace` ➡ `Datadog.Trace.Tests`
- `Datadog.Trace.OpenTracing` ➡ `Datadog.Trace.OpenTracing.Tests`
- `Datadog.Trace.ClrProfiler.Managed` ➡️ `Datadog.Trace.ClrProfiler.Managed.Tests`

But I thought this was `Datadog.Trace.ClrProfiler.IntegrationTests`, so never mind. And maybe having so many individual test assemblies doesn't scale so well (106 projects!).

Reply:
> I'm used to having one test assembly for every assembly

Me too, + optional integration test project, which is what we have 😄 The real problem with the number of projects is all the sample projects 🤷